docs/Rest-API.md

```diff
@@ -0,0 +1,147 @@
+# Axe REST API
+
+Documentation of the REST API that comes along with the Axe service.
+
+## AxeResult
+
+This endpoint serves as a wrapper to collect all of the information from an axe run. 
+
+### Motivation
+
+- Provide access to results and context easily.
+- Avoid timing issues that come with multiple requests.
+
+### Simplified Sample
+```
+{
+  "axeContext" : { @AxeContext },
+  "axeResult" : { AXRL },
+  "axeConf" : { TBD by Framework }  
+}
+```
+
+### Requirements
+
+- Data points are collected at the same point in time, or as close as can be expected.
+
+## AxeContext
+
+Wraps data together that is needed to identify a particular View, with a particular screenshot,
+for a particular device configuration.
+
+### Motivation
+
+- Collect View information in one place.
+- Allow for direct correlation of screenshot and ViewHierarchy
+- Easy developer debugging.
+- Reproducability of results.
+
+### Simplified Sample
+
+```
+{
+  "axeView" : @AxeView,
+  "axeDevice" : @AxeDevice,
+  "axeEvents" : @AxeEvents,
+  "screenshot" : "base64png data"
+}
+```
+
+## AxeView
+
+Serves up properties about the Accessibility Hierarchy of a particular view. 
+
+### Motivation
+
+- All motivations of AxeContext apply.
+- Rule debugging.
+- Simplified test case collection.
+- Simplified false positive identification.
+
+### Simplified Sample
+
+```
+{
+  "Arbitrary Property Name": "value1",
+  "Arbitrary Property Name2": "value2",
+  "children" : [
+    {
+      "Arbitrary Property Name": "value3",
```

Comment:
>"Arbitrary Property Name": "value3", [](start = 6, length = 36)

A comment more about the design than the document: A string to string mapping makes for excellent portability for the data generator, but horrible portability for the data consumer. For example, it's likely that Android, iOS, and Windows will have multiple overlapping concepts as properties, but with potentially different property names and/or data formats. In order to create a single viewer that can support all of the variants, we'll probably need to convert the platform-specific data into a platform-agnostic format to keep the view code sane